spec/Section 3 -- Type System.md

```diff
@@ -818,6 +818,9 @@ of rules must be adhered to by every Object type in a GraphQL schema.
          characters {"__"} (two underscores).
       2. The argument must accept a type where {IsInputType(argumentType)}
          returns {true}.
+      3. If the argument has a default value, {defaultValue} must be compatible
+         with {argumentType} as per the coercion rules for that type, and
+         coercion of {defaultValue} must not cause an infinite loop.
```

Comment:
I am not sure we should state `not cause an infinite loop`: this is a general requirement,right? :-)

Reply:
Letting people know of the risk is important ;) But seriously, this is the part I need to pick Lee's brains on, though if the RFC @IvanGoncharov referenced is merged then infinite recursion in defaultValues will be prevented elsewhere, and thus this won't be needed.

Reply:
I think we must clearly distinguish several aspects here: I agree that `defaultValue` must be coerced.  
And I also agree that in the case of input objects there is the possibility of default values causing infinite recursion if evaluated statically. But this is a special case for input objects coercion and we should deal with it separately from the default value coercion (this doesn't mean separate PR, but just two problems).

Reply:
Agreed; I think this needs to be a separate validation rule. This text is just a placeholder to remind us something needs to be written about this problem, I never intended this part to make it into the spec hence needing to pick Lee’s brains on the right approach. https://github.com/graphql/graphql-spec/pull/793#issuecomment-738736539

Reply:
An argument can never cause an infinite loop anyway; so I've removed this - but I've also fixed the "must not cause an infinite loop" text that was below too.